amplify/backend/function/fetchMotionState/src/utils.js

```diff
@@ -0,0 +1,36 @@
+const { providers } = require('ethers');
+const {
+  getColonyNetworkClient,
+  Network,
+  Extension,
+} = require('@colony/colony-js');
+
+const RPC_URL = 'http://network-contracts.docker:8545'; // this needs to be extended to all supported networks
+const {
+  etherRouterAddress: networkAddress,
+} = require('../../../../mock-data/colonyNetworkArtifacts/etherrouter-address.json');
+
+const getNetworkClient = () => {
+  const provider = new providers.JsonRpcProvider(RPC_URL);
+
+  const networkClient = getColonyNetworkClient(Network.Custom, provider, {
+    networkAddress,
+    reputationOracleEndpoint:
+      'http://reputation-monitor.docker:3001/reputation/local',
+  });
+
+  return networkClient;
+};
+
+const getVotingClient = async (colonyAddress) => {
+  const networkClient = getNetworkClient();
+  const colonyClient = await networkClient.getColonyClient(colonyAddress);
+  return colonyClient.getExtensionClient(Extension.VotingReputation);
+};
+
+const getLatestMotionState = async (colonyAddress, motionId) => {
+  const votingReputationClient = await getVotingClient(colonyAddress);
+  return votingReputationClient.getMotionState(motionId);
+};
```

Comment:
Is there no way (ie. event that gets emitted when the state changes) to cache that on the motion model directly?

Reply:
The problem is when the timer runs out. The contracts don't have a concept of chron-jobs, therefore no event gets emitted when the timer runs out. Instead, you have to query the motion state on the contracts directly.